test/e2e/specs/specs-wpcom/wp-gutenberg__experimental.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * @group gutenberg
+ */
+
+import {
+	setupHooks,
+	TestAccount,
+	BrowserHelper,
+	DataHelper,
+	GutenbergEditorPage,
+} from '@automattic/calypso-e2e';
+import { Page } from 'playwright';
+
+describe( DataHelper.createSuiteTitle( 'Gutenberg: Experimental Features' ), function () {
+	const accountName = BrowserHelper.targetGutenbergEdge()
+		? 'gutenbergSimpleSiteEdgeUser'
+		: 'gutenbergSimpleSiteUser';
+
+	let page: Page;
+	let gutenbergEditorPage: GutenbergEditorPage;
+
+	setupHooks( async ( args ) => {
+		page = args.page;
+
+		const testAccount = new TestAccount( accountName );
+		await testAccount.authenticate( page );
+	} );
+
+	it( 'Go to the new post page', async function () {
+		gutenbergEditorPage = new GutenbergEditorPage( page );
+		await gutenbergEditorPage.visit( 'post' );
+	} );
+
+	it.each( [
+		[ 'blockEditor', '__unstableInserterMenuExtension', 'function' ],
+		[ 'date', '__experimentalGetSettings', 'function' ],
+		[ 'components', '__experimentalNavigationBackButton', 'object' ],
+		[ 'editPost', '__experimentalMainDashboardButton', 'function' ],
+	] )(
+		'Experimental package %s and feature %s are available',
+		async function ( packageName, feature, featureType ) {
+			const frame = await gutenbergEditorPage.getEditorFrame();
+			const packageAvailable = await frame.evaluate( `typeof window[ "wp" ]["${ packageName }"]` );
+
+			expect( packageAvailable ).not.toStrictEqual( 'undefined' );
+			// It should always be an object.
+			expect( packageAvailable ).toStrictEqual( 'object' );
+
+			const featureAvailable = await frame.evaluate(
+				`typeof window[ "wp" ]["${ packageName }"]["${ feature }"]`
+			);
+			expect( featureAvailable ).not.toStrictEqual( 'undefined' );
+			// Feature type changes between object and function and depends on the feature itself.
+			expect( featureAvailable ).toContain( featureType );
+		}
+	);
+
+	it( 'Experimental data is available', async function () {
+		const frame = await gutenbergEditorPage.getEditorFrame();
+		const blockPatterns = await frame.evaluate(
+			`Array.isArray( window.wp.data.select( 'core/editor' ).getEditorSettings().__experimentalBlockPatterns )`
+		);
+
+		expect( blockPatterns ).toStrictEqual( true );
```

Comment:
@fullofcaffeine This assertion was previously as follows:
```
			assert(
				__experimentalBlockPatternsAreIterable,
				'__experimentalBlockPatterns was not iterable, please contact #team-ganon to update premium pattern highlighting'
			);
```

Despite running the command `Array.isArray( window.wp.data.select( 'core/editor' ).getEditorSettings().__experimentalBlockPatterns )` with numerous accounts (including my personal account) I cannot get the originally expected output. Each time, the command returns `true` as the output.

As a result, I've changed the assertion to expect the value `true` instead. Could you verify that this change is acceptable?

Reply:
> Despite running the command Array.isArray( window.wp.data.select( 'core/editor' ).getEditorSettings().__experimentalBlockPatterns ) with numerous accounts (including my personal account) I cannot get the originally expected output. Each time, the command returns true as the output.

Hmm, it looks like the previous `assert` was also checking for `true`. The second param is just a custom message in case of failure.

Reply:
Here's the jsdoc for `assert`:

```
(alias) assert(value: unknown, message?: string | Error): asserts value
import assert
An alias of ok .

@since — v0.5.9

@param value — The input that is checked for being truth.
```

Reply:
I think it'd still be nice to pass that same custom message, as in:

```js
expect( blockPatterns, '__experimentalBlockPatterns was not iterable, please contact #team-ganon to update premium pattern highlighting' ).toStrictEqual( true );
```

Reply:
Ah nevermind. Jest doesn't support this out-of-the-box, it actually needs an extension for that, it seems (https://github.com/mattphillips/jest-expect-message) :/. I'll just add a comment above the test with the same text. I think keeping the directions there about contacting team-ganon is important.

Reply:
Got it, thanks. I understand the second text string in the original assertion is the error message in case the assertion fails. 

The question was more about what the expected value of `__experimentalBlockPatternsAreIterable` was but you've helped answer that for me. Thanks!

Reply:
Oh, cool! Yeah, `true` is fine :)